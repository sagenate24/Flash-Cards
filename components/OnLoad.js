import React, { Component } from 'react';
import {
  Text,
  View,
  StyleSheet,
  Animated,
  StatusBar
} from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

const AnimatedIcon = Animated.createAnimatedComponent(MaterialCommunityIcons);

class OnLoad extends Component {
  state = {
    opacity: new Animated.Value(0),
    bounceValue: new Animated.Value(0.1),
  };

  componentDidMount() {
    const {
      opacity,
      bounceValue,
    } = this.state;

    Animated.timing(opacity, {
      toValue: 1,
      duration: 800,
    }).start();
    Animated.sequence([
      Animated.delay(300),
      Animated.timing(bounceValue, {
        duration: 200,
        toValue: 1.04
      }),
      Animated.spring(bounceValue, {
        toValue: 1,
        friction: 4
      }),
    ]).start();
  };

  render() {
    const {
      opacity,
      bounceValue
    } = this.state;

    return (
      <View style={styles.container}>
        <StatusBar hidden={true}/>
        <View>
          <AnimatedIcon
            size={100}
            name='cards-outline'
            style={[styles.logo, { opacity, transform: [{scale: bounceValue}] }]} />
        </View>
        <View>
          <Animated.Text
            style={[styles.textLogo, {
              opacity,
              transform: [{scale: bounceValue}],
            }]}>
            FlashCards
          </Animated.Text>
        </View>
      </View>
    );
  };
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#4e8098',
  },
  logo: {
    alignItems: 'center',
    color: '#fff'
  },
  textLogo: {
    fontSize: 26,
    fontWeight: 'bold',
    color: '#fff',
    alignItems: 'center',
  },
});

export default OnLoad;
