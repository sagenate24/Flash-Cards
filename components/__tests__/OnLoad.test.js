import React from 'react';
import { shallow } from 'enzyme';
import { Animated } from 'react-native';

import OnLoad from '../OnLoad';

jest.mock('Animated');

beforeEach(() => {
  Animated.timing.mockImplementation(() => ({ start: jest.fn(), stop: jest.fn() }));
  Animated.sequence.mockImplementation(() => ({ start: jest.fn(), stop: jest.fn() }));
  Animated.spring.mockImplementation(() => ({ start: jest.fn(), stop: jest.fn() }));
});


it('should match Snapshot', (done) => {
  const snap = shallow(<OnLoad />);

  expect(snap).toMatchSnapshot();

  done();
});
