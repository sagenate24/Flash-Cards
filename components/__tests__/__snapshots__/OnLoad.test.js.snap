// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OnLoad />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          hidden={true}
          showHideTransition="fade"
        />,
        <Component>
          <UNDEFINED
            name="cards-outline"
            size={100}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                },
                Object {
                  "opacity": undefined,
                  "transform": Array [
                    Object {
                      "scale": undefined,
                    },
                  ],
                },
              ]
            }
          />
        </Component>,
        <Component>
          <AnimatedComponent
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontSize": 26,
                  "fontWeight": "bold",
                },
                Object {
                  "opacity": undefined,
                  "transform": Array [
                    Object {
                      "scale": undefined,
                    },
                  ],
                },
              ]
            }
          >
            FlashCards
          </AnimatedComponent>
        </Component>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#4e8098",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "hidden": true,
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <UNDEFINED
            name="cards-outline"
            size={100}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                },
                Object {
                  "opacity": undefined,
                  "transform": Array [
                    Object {
                      "scale": undefined,
                    },
                  ],
                },
              ]
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "cards-outline",
            "size": 100,
            "style": Array [
              Object {
                "alignItems": "center",
                "color": "#fff",
              },
              Object {
                "opacity": undefined,
                "transform": Array [
                  Object {
                    "scale": undefined,
                  },
                ],
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <AnimatedComponent
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontSize": 26,
                  "fontWeight": "bold",
                },
                Object {
                  "opacity": undefined,
                  "transform": Array [
                    Object {
                      "scale": undefined,
                    },
                  ],
                },
              ]
            }
          >
            FlashCards
          </AnimatedComponent>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "FlashCards",
            "style": Array [
              Object {
                "alignItems": "center",
                "color": "#fff",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              Object {
                "opacity": undefined,
                "transform": Array [
                  Object {
                    "scale": undefined,
                  },
                ],
              },
            ],
          },
          "ref": null,
          "rendered": "FlashCards",
          "type": [MockFunction],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            hidden={true}
            showHideTransition="fade"
          />,
          <Component>
            <UNDEFINED
              name="cards-outline"
              size={100}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#fff",
                  },
                  Object {
                    "opacity": undefined,
                    "transform": Array [
                      Object {
                        "scale": undefined,
                      },
                    ],
                  },
                ]
              }
            />
          </Component>,
          <Component>
            <AnimatedComponent
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#fff",
                    "fontSize": 26,
                    "fontWeight": "bold",
                  },
                  Object {
                    "opacity": undefined,
                    "transform": Array [
                      Object {
                        "scale": undefined,
                      },
                    ],
                  },
                ]
              }
            >
              FlashCards
            </AnimatedComponent>
          </Component>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#4e8098",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "hidden": true,
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <UNDEFINED
              name="cards-outline"
              size={100}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#fff",
                  },
                  Object {
                    "opacity": undefined,
                    "transform": Array [
                      Object {
                        "scale": undefined,
                      },
                    ],
                  },
                ]
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "cards-outline",
              "size": 100,
              "style": Array [
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                },
                Object {
                  "opacity": undefined,
                  "transform": Array [
                    Object {
                      "scale": undefined,
                    },
                  ],
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": undefined,
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <AnimatedComponent
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#fff",
                    "fontSize": 26,
                    "fontWeight": "bold",
                  },
                  Object {
                    "opacity": undefined,
                    "transform": Array [
                      Object {
                        "scale": undefined,
                      },
                    ],
                  },
                ]
              }
            >
              FlashCards
            </AnimatedComponent>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "FlashCards",
              "style": Array [
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontSize": 26,
                  "fontWeight": "bold",
                },
                Object {
                  "opacity": undefined,
                  "transform": Array [
                    Object {
                      "scale": undefined,
                    },
                  ],
                },
              ],
            },
            "ref": null,
            "rendered": "FlashCards",
            "type": [MockFunction],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
